fem/tmop/tmop_pa_jt3_datc.cpp

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2010-2021, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "../tmop.hpp"
+#include "tmop_pa.hpp"
+#include "../../general/forall.hpp"
+#include "../../linalg/kernels.hpp"
+
+namespace mfem
+{
+
+MFEM_REGISTER_TMOP_KERNELS(bool, DatcSize,
+                           const int NE,
+                           const int ncomp,
+                           const int sizeidx,
+                           const DenseMatrix w_, // Copy
```

Comment:
Why copy?

Reply:
I think it was to avoid copying the `Geometry` class as it comes from `const DenseMatrix &W = Geometries.GetGeomToPerfGeomJac(fe.GetGeomType());`

Reply:
I was wondering the same.

Reply:
Wait, to AVOID copy you want to use `const DenseMatrix &w_` right? Or am I missing something?

Reply:
I needed the copy at some point to avoid pulling any object on the device

Reply:
but I can't reproduce the error - I'll avoid the copy then.

Reply:
`done`

Reply:
It turns out that the static `DenseMatrix` is initialized with `HOST` MemoryType. The Read with the `debug` and the `DEFAULT` setup will try to use the `DEVICE` memory controller.

Reply:
Addressed by changing the dual map when using the `debug` device. Thanks @v-dobrev !